lib/plugins/azurepush.js

```diff
@@ -0,0 +1,323 @@
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x08-Failed to verify index on alert in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.info', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x09-Failed to verify index on info in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      
+      // More info on request body: https://msdn.microsoft.com/en-us/library/mt621153.aspx
+      var installation = {
+        installationId: request.installationId
+        , pushChannel: request.deviceToken
+        , templates: {
+          nightscout: {
+            tags: tags
+          }
+        }
+      };
+
+      // 'platform': https://developer.xamarin.com/guides/xamarin-forms/platform-features/device/#Device.OS
+      switch (request.platform) {
+        case 'iOS':
+          installation.platform = 'apns';
+          installation.templates.nightscout.body = '{\"aps\":{\"alert\":\"$(message)\",\"sound\":\"$(sound)\",\"category\":\"$(category)\"},\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"title\":\"$(title)\"}';
+          break;
+        case 'Android':
+          installation.platform = 'gcm';
+          installation.templates.nightscout.body = '{\"data\":{\"message\":\"$(message)\",\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"sound\":\"$(sound)\",\"title\":\"$(title)\"}}';
+          break;
+        case 'WinPhone': //Not yet supported
+        case 'Windows': //Not yet supported
+        default: //No platform specified, or invalid platform
+          // *****REMOVE*****: Beta logging
+          logTrelloIssue(env, '0x10-Register received without supported platform');
+          // END: *****REMOVE*****
+
+          return callback({message: 'Platform not supported'});
+      }
+
+      notificationHubService.createOrUpdateInstallation(installation, function (error, response) {
+        // *****REMOVE*****: Beta logging
+        if (error) {
+          logTrelloIssue(env, '0x11-Failed to create or update installation');
+        }
+        // END: *****REMOVE*****
+
+        callback(error, {installationId: installation.installationId});
+      });
+    }
+    else {
+      // *****REMOVE*****: Beta logging
+      logTrelloIssue(env, '0x12-Register received without platform');
+      // END: *****REMOVE*****
+
+      // No platform, required parameter
+      return callback({message: 'Platform required'});
+    }
+  };
+
+  azurepush.ack = function ack(request, callback) {
+    if (request && request.level) {
```

Comment:
Yeah, currently snoozing is open, and I really want to fix that now that we have a permissions system using the tokens.  I started to implement `notification:*.ack`, but it's not done.  A notification id would be great, I think that's the direction we need to go.  Also I think there's a difference between a local snooze and a system wide ack.  Anyone should be able to make the alarm they are hearing locally stop, but there should be a limit on how that effects others using the system.

If we can't find a way to lock it down more now we should only add the endpoint if the plugin is enabled so we don't add a new hole by default.